src/main/java/de/hhu/propra/link/controllers/LinkController.java

```diff
@@ -82,4 +88,72 @@ private void setMessages(String errorMessage, String successMessage) {
         this.errorMessage = errorMessage;
         this.successMessage = successMessage;
     }
+
+    /**
+     * creates a not yet used viable abbrevation for the entered url
+     * @param url The URL that is to be shortened
+     * @return the abbrevation for the long URL
+     */
+    private String makeAbbrevation(String url){
+
+        // removing the scheme from the URL
+        // "https://example.com" -> "example.com"
+        int schemelessUrlindex = url.indexOf("://");
+        String schemelessUrl = url;
+        if(schemelessUrlindex > 0) {
+            schemelessUrl = url.substring(schemelessUrlindex + 3, url.length());
+        } else
+            return url; // URL possibly not correct, so just return it
+
+        // determine a suitable abbreviation taking the paths in consideration
+        // ex: "sub.example.com/path/to/index.html" -> "sbxmplpti"
+        String[] domainAndPaths = schemelessUrl.split("/");
+        String tmp_abbreviation = unvowelize(domainAndPaths[0]);
+        for(int i = 1; i < domainAndPaths.length; i++){
+            tmp_abbreviation += domainAndPaths[i].charAt(0);
+        }
+
+        // create another abbreviation if the preferred one already exists
+        int i = 1;
+        String abbrevation = tmp_abbreviation;
+        while( ! linkRepository.findById(abbrevation).isEmpty() ){
+            abbrevation = new String(tmp_abbreviation + i);
+            i += 1;
+        }
+
+        return abbrevation;
+    }
+
+    /**
+     * @param domain The domain name where the vowels, dots and the TLD should be removed from
+     * @return the String without any vowels, dots and the TLD
+     */
+    private String unvowelize(String domain) {
+        String unvowelized = "";
+
+        for(int i = 0; i < domain.lastIndexOf('.'); i++){
+            if(! (isVowel(domain.charAt(i)) || domain.charAt(i) == '.') ){
+                unvowelized += domain.charAt(i);
+            }
+        }
+
+        return unvowelized;
+    }
+
+    /**
+     * returns true if entered letter is a vowel, else false.
```

Comment:
Please respect the English Grammar. You are writing a sentence and a sentence starts with a Capital Character.